File: CS/Dashboard_EntityFramework/Form1.cs
using DevExpress.DashboardCommon;
using DevExpress.XtraEditors;
using DevExpress.DataAccess.EntityFramework;

namespace Dashboard_EntityFramework {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();
            InitializeDashboard();          
        }

        public void InitializeDashboard() {
            Dashboard dashboard = new Dashboard();
            DashboardEFDataSource efDataSource = new DashboardEFDataSource();
            efDataSource.ConnectionParameters =
                new EFConnectionParameters(typeof(OrdersContext));
            dashboard.DataSources.Add(efDataSource);

            PivotDashboardItem pivot = new PivotDashboardItem();
            pivot.DataMember = "Orders";
            pivot.DataSource = dashboard.DataSources[0];
            pivot.Rows.AddRange(new Dimension("ShipCountry"), new Dimension("ShipCity"));
            pivot.Columns.Add(new Dimension("OrderDate"));
            pivot.Values.Add(new Measure("Freight"));

            ChartDashboardItem chart = new ChartDashboardItem();
            chart.DataSource = dashboard.DataSources[0];
            chart.DataMember = "Orders";
            chart.Arguments.Add(new Dimension("OrderDate", DateTimeGroupInterval.Year));
            chart.Panes.Add(new ChartPane());
            SimpleSeries freightSeries = new SimpleSeries(SimpleSeriesType.Bar);
            freightSeries.Value = new Measure("Freight");
            chart.Panes[0].Series.Add(freightSeries);

            dashboard.Items.AddRange(pivot, chart);
            dashboardViewer1.Dashboard = dashboard;
        }
    }
}

File: VB/Dashboard_EntityFramework/Form1.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.XtraEditors
Imports DevExpress.DataAccess.EntityFramework

Namespace Dashboard_EntityFramework
	Partial Public Class Form1
		Inherits XtraForm

		Public Sub New()
			InitializeComponent()
			InitializeDashboard()
		End Sub

		Public Sub InitializeDashboard()
			Dim dashboard As New Dashboard()
			Dim efDataSource As New DashboardEFDataSource()
			efDataSource.ConnectionParameters = New EFConnectionParameters(GetType(OrdersContext))
			dashboard.DataSources.Add(efDataSource)

			Dim pivot As New PivotDashboardItem()
			pivot.DataMember = "Orders"
			pivot.DataSource = dashboard.DataSources(0)
			pivot.Rows.AddRange(New Dimension("ShipCountry"), New Dimension("ShipCity"))
			pivot.Columns.Add(New Dimension("OrderDate"))
			pivot.Values.Add(New Measure("Freight"))

			Dim chart As New ChartDashboardItem()
			chart.DataSource = dashboard.DataSources(0)
			chart.DataMember = "Orders"
			chart.Arguments.Add(New Dimension("OrderDate", DateTimeGroupInterval.Year))
			chart.Panes.Add(New ChartPane())
			Dim freightSeries As New SimpleSeries(SimpleSeriesType.Bar)
			freightSeries.Value = New Measure("Freight")
			chart.Panes(0).Series.Add(freightSeries)

			dashboard.Items.AddRange(pivot, chart)
			dashboardViewer1.Dashboard = dashboard
		End Sub
	End Class
End Namespace
